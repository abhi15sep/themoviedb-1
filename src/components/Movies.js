import React from 'react';
import PropTypes from 'prop-types'
import Movie from './Movie';
import { Grid, Row } from 'react-bootstrap';

function Movies(props) {
  const movies = props.data.map((movie) => {
    return <Movie key={movie.id} data={movie} />;
  });

  return (
    <Grid>
      <Row>
        {movies}
      </Row>
    </Grid>
  );
}

Movies.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default Movies;
