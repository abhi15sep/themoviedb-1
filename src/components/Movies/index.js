import React from 'react';
import PropTypes from 'prop-types';
import { Container, Row } from 'react-bootstrap';
import Movie from '../Movie';
import Types from '../../utils/types';
import './styles.css';

/**
 * Component for display the movie list
 *
 * @param {object} props - The props of the component
 * @param {object} props.movies - The movie list
 * @returns {Movies} - The react component
 */
export default function Movies({ movies }) {
  return (
    <Container className="Movies-main">
      <Row>
        {movies.map((movie) => <Movie key={movie.id} movie={movie} />)}
      </Row>
    </Container>
  );
}

Movies.propTypes = {
  movies: PropTypes.arrayOf(Types.movie.isRequired).isRequired,
};
