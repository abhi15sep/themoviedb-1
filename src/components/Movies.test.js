import React from 'react';
import ReactDOM from 'react-dom';
import Movies from './Movies';

const movies = [
    {
      id: 1,
      vote_average: 5,
      title: 'example 1'
    },
    {
      id: 2,
      vote_average: 5,
      title: 'example 2'
    },
    {
      id: 3,
      vote_average: 9,
      title: 'example 3'
    }
  ];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Movies data={movies} />, div);
  ReactDOM.unmountComponentAtNode(div);
});
