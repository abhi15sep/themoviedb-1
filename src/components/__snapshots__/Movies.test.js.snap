// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-6 col-sm-12"
    >
      <div
        className="Movie-card"
      >
        <div
          className="row"
        >
          <div
            className="col-md-4 col-sm-12"
          >
            <a
              href="/details/1"
              onClick={[Function]}
            >
              <img
                alt="example 1"
                src="http://image.tmdb.org/t/p/w185/path/to/image"
                title="example 1"
              />
            </a>
          </div>
          <div
            className="col-md-8 col-sm-12"
          >
            <div
              className="Movie-body"
            >
              <div
                className="media"
              >
                <div
                  className="media-left"
                >
                  <span
                    className="label label-danger"
                  >
                    5
                  </span>
                </div>
                <div
                  className="media-body"
                >
                  <h4
                    className="media-heading"
                  >
                    <a
                      href="/details/1"
                      onClick={[Function]}
                    >
                      <span
                        className="Movie-title"
                      >
                        example 1
                      </span>
                    </a>
                    <br />
                    <small>
                      April 28, 2017
                    </small>
                  </h4>
                </div>
              </div>
              <div
                className="Movie-overview"
              >
                <p>
                  Test
                  ...
                </p>
              </div>
              <div
                className="Movie-details"
              >
                <a
                  className="Movie-details-link"
                  href="/details/1"
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="glyphicon glyphicon-info-sign"
                  />
                    More info
                </a>
                <div
                  className="Recommendation-main"
                >
                  <span
                    aria-hidden="true"
                    className="glyphicon glyphicon-thumbs-down"
                  />
                   
                  <small>
                    Bad choice
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-lg-6 col-sm-12"
    >
      <div
        className="Movie-card"
      >
        <div
          className="row"
        >
          <div
            className="col-md-4 col-sm-12"
          >
            <a
              href="/details/2"
              onClick={[Function]}
            >
              <img
                alt="example 2"
                src="http://image.tmdb.org/t/p/w185/path/to/image"
                title="example 2"
              />
            </a>
          </div>
          <div
            className="col-md-8 col-sm-12"
          >
            <div
              className="Movie-body"
            >
              <div
                className="media"
              >
                <div
                  className="media-left"
                >
                  <span
                    className="label label-danger"
                  >
                    3
                  </span>
                </div>
                <div
                  className="media-body"
                >
                  <h4
                    className="media-heading"
                  >
                    <a
                      href="/details/2"
                      onClick={[Function]}
                    >
                      <span
                        className="Movie-title"
                      >
                        example 2
                      </span>
                    </a>
                    <br />
                    <small>
                      April 28, 2017
                    </small>
                  </h4>
                </div>
              </div>
              <div
                className="Movie-overview"
              >
                <p>
                  Test
                  ...
                </p>
              </div>
              <div
                className="Movie-details"
              >
                <a
                  className="Movie-details-link"
                  href="/details/2"
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="glyphicon glyphicon-info-sign"
                  />
                    More info
                </a>
                <div
                  className="Recommendation-main"
                >
                  <span
                    aria-hidden="true"
                    className="glyphicon glyphicon-thumbs-down"
                  />
                   
                  <small>
                    Bad choice
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-lg-6 col-sm-12"
    >
      <div
        className="Movie-card"
      >
        <div
          className="row"
        >
          <div
            className="col-md-4 col-sm-12"
          >
            <a
              href="/details/3"
              onClick={[Function]}
            >
              <img
                alt="example 3"
                src="http://image.tmdb.org/t/p/w185/path/to/image"
                title="example 3"
              />
            </a>
          </div>
          <div
            className="col-md-8 col-sm-12"
          >
            <div
              className="Movie-body"
            >
              <div
                className="media"
              >
                <div
                  className="media-left"
                >
                  <span
                    className="label label-success"
                  >
                    9
                  </span>
                </div>
                <div
                  className="media-body"
                >
                  <h4
                    className="media-heading"
                  >
                    <a
                      href="/details/3"
                      onClick={[Function]}
                    >
                      <span
                        className="Movie-title"
                      >
                        example 3
                      </span>
                    </a>
                    <br />
                    <small>
                      April 28, 2017
                    </small>
                  </h4>
                </div>
              </div>
              <div
                className="Movie-overview"
              >
                <p>
                  Test
                  ...
                </p>
              </div>
              <div
                className="Movie-details"
              >
                <a
                  className="Movie-details-link"
                  href="/details/3"
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="glyphicon glyphicon-info-sign"
                  />
                    More info
                </a>
                <div
                  className="Recommendation-main"
                >
                  <span
                    aria-hidden="true"
                    className="glyphicon glyphicon-thumbs-up"
                  />
                   
                  <small>
                    Editor choice
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
